### PR TITLE
 fix #2521: Wrong VRam value 

### LINKED ISSUE #2521: Wrong VRam value
Trying detect graphic card parameters, but got wrong info:
Os family - Windows
Os bitness - 64
Graphics card found: 2
GPU - NVIDIA (0x10de)/NVIDIA GeForce RTX 4090/4293918720
GPU - Intel Corporation (0x8086)/Intel(R) UHD Graphics 770/1073741824

VRam size = 4293918720 = UINT.MAX_VALUE - 1MiB
Can't find info why I get that result.
Looks like type conversion issue.

### PR DESCRIPTION
**Pull request by komelgman**
fix #2521: 
Changed GPU info gathering mecahism for Windows.
Using Win Registy values, instead of Win32 api calls, so now vram more than 4GiB can be detected properly

### CODE REVIEW

File: oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsGraphicsCard.java [hunk 1]
```diff
@@ -49,6 +66,76 @@ final class WindowsGraphicsCard extends AbstractGraphicsCard {
...
+                String vendor = Advapi32Util.registryGetStringValue(WinReg.HKEY_LOCAL_MACHINE, fullKey, VENDOR);
+                String versionInfo = Advapi32Util.registryGetStringValue(WinReg.HKEY_LOCAL_MACHINE, fullKey, DRIVER_VERSION);
+                long vram = 0L;
+
+                if (Advapi32Util.registryValueExists(WinReg.HKEY_LOCAL_MACHINE, fullKey, QW_MEMORY_SIZE)) {
+                    vram = Advapi32Util.registryGetLongValue(WinReg.HKEY_LOCAL_MACHINE, fullKey, QW_MEMORY_SIZE);
+                } else if (Advapi32Util.registryValueExists(WinReg.HKEY_LOCAL_MACHINE, fullKey, MEMORY_SIZE)) {
+                    Object genericValue = Advapi32Util.registryGetValue(WinReg.HKEY_LOCAL_MACHINE, fullKey, MEMORY_SIZE);
+                    if (genericValue instanceof Long || genericValue instanceof Integer) {
+                        vram = (long) genericValue;
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsGraphicsCard.java:93]: I think you need to handle `Long` and `Integer` separately as the int will be an unsigned DWORD.
```suggestion
                    if (genericValue instanceof Long) {
                        vram = genericValue.longValue();
                    else if (genericValue instanceof Integer) {
                        vram = Integer.toUnsignedLong(genericValue.intValue());
```

**Review by dbwiddis — COMMENTED** [review]
Thanks for this contribution!  Looks great, a few suggestions to make it better.

**Comment by dbwiddis** [comment]
Also can you add an entry to the CHANGELOG for this?

File: oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsGraphicsCard.java [hunk 2]
```diff
@@ -49,6 +66,76 @@ final class WindowsGraphicsCard extends AbstractGraphicsCard {
...
+                    } else if (genericValue instanceof byte[]) {
+                        vram = memSizeBinaryToLong((byte[]) genericValue);
```
[the remark this answers is not in the document]
Comment (komelgman) [reply oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsGraphicsCard.java:95]: Should be
```java
vram = ParseUtil.byteArrayToLong(bytes, bytes.length, false);
```

**Comment by komelgman** [comment]
Sorry, I'm not very skilled in github PR workflow, should I make suggested checnges on my side by hand, and update PR?

**Comment by dbwiddis** [comment]
> Sorry, I'm not very skilled in github PR workflow, should I make suggested checnges on my side by hand, and update PR?

Just make a new commit and push it, it will update this PR

**Review by dbwiddis — APPROVED** [review]
LGTM! Will merge when CI completes.

Thanks for this fix!

File: oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsGraphicsCard.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (dbwiddis) [reply oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsGraphicsCard.java]: @komelgman I guess WIndows likes little-endianness.

**Pull request merged**